### PR TITLE
Server libs

### PR DESCRIPTION
**Pull request by tristantarrant**
https://github.com/infinispan/infinispan-images/issues/156
Requires https://github.com/infinispan/infinispan/pull/9676

### CODE REVIEW

File: README.md
Comment (ryanemerson) [inline README.md:126]: This results in an error for me:

```
Could not download 'org/postgresql/postgresql/42.3.1/postgresql-42.3.1' from 'https://repo1.maven.org/maven2/' repository
IllegalArgumentException: Artifac 'org.postgresql:postgresql:42.3.1' not found

```

File: modules/org/infinispan/runtime/added/bin/launch.sh
Comment (ryanemerson) [inline modules/org/infinispan/runtime/added/bin/launch.sh:64]: We should probably check the return code of the process here and quit if the CLI execution fails. Without this, the server will continue to start when the dependencies can't be resolved.

File: modules/org/infinispan/runtime/added/bin/launch.sh
Comment (ryanemerson) [inline modules/org/infinispan/runtime/added/bin/launch.sh:141]: Changes to the quotes here break the image. This is because of a bug in `server.sh`, the following `./server.sh "--bind-address=0.0.0.0 "` results in server startup failing with:

```java
2021-11-22 14:55:55,195 INFO  (main) [BOOT] JVM OpenJDK 64-Bit Server VM Red Hat, Inc. 11.0.13+8
2021-11-22 14:55:55,205 INFO  (main) [BOOT] JVM arguments = [-server, -Xlog:gc*:file=/tmp/infinispan-server-14.0.0-SNAPSHOT/server/log/gc.log:time,uptimemillis:filecount=5,filesize=3M, -Xms64m, -Xmx512m, -XX:MetaspaceSize=64M, -Djava.net.preferIPv4Stack=true, -Djava.awt.headless=true, -Dvisualvm.display.name=infinispan-server, -Djava.util.logging.manager=org.apache.logging.log4j.jul.LogManager, -Dinfinispan.server.home.path=/tmp/infinispan-server-14.0.0-SNAPSHOT, -classpath, :/tmp/infinispan-server-14.0.0-SNAPSHOT/boot/infinispan-server-runtime-14.0.0-SNAPSHOT-loader.jar, org.infinispan.server.loader.Loader, org.infinispan.server.Bootstrap, --bind-address=0.0.0.0 ]
2021-11-22 14:55:55,210 INFO  (main) [BOOT] PID = 268595
2021-11-22 14:55:55,235 INFO  (main) [org.infinispan.SERVER] ISPN080000: Infinispan Server starting
2021-11-22 14:55:55,235 INFO  (main) [org.infinispan.SERVER] ISPN080017: Server configuration: infinispan.xml
2021-11-22 14:55:55,236 INFO  (main) [org.infinispan.SERVER] ISPN080032: Logging configuration: /tmp/infinispan-server-14.0.0-SNAPSHOT/server/conf/log4j2.xml
2021-11-22 14:55:55,782 FATAL (main) [org.infinispan.SERVER] ISPN080028: Infinispan Server failed to start org.infinispan.commons.CacheConfigurationException: ISPN080033: Cannot find a network address which matches the supplied configuration
	at org.infinispan.server.configuration.ServerConfigurationParser.parseInterface(ServerConfigurationParser.java:209)
	at org.infinispan.server.configuration.ServerConfigurationParser.parseInterfaces(ServerConfigurationParser.java:196)
	at org.infinispan.server.configuration.ServerConfigurationParser.parseServerElements(ServerConfigurationParser.java:115)
	at org.infinispan.server.configuration.ServerConfigurationParser.readElement(ServerConfigurationParser.java:101)
	at org.infinispan.configuration.parsing.ParserRegistry.parseElement(ParserRegistry.java:204)
	at org.infinispan.configuration.parsing.ConfigurationBuilderHolder.handleAnyElement(ConfigurationBuilderHolder.java:160)
	at org.infinispan.commons.configuration.io.AbstractConfigurationReader.handleAny(AbstractConfigurationReader.java:57)
	at org.infinispan.configuration.parsing.Parser.readElement(Parser.java:98)
	at org.infinispan.configuration.parsing.ParserRegistry.parseElement(ParserRegistry.java:204)
	at org.infinispan.configuration.parsing.ParserRegistry.parse(ParserRegistry.java:182)
	at org.infinispan.configuration.parsing.ParserRegistry.parse(ParserRegistry.java:170)
	at org.infinispan.configuration.parsing.ParserRegistry.parse(ParserRegistry.java:164)
	at org.infinispan.server.Server.parseConfiguration(Server.java:306)
	at org.infinispan.server.Server.<init>(Server.java:224)
	at org.infinispan.server.Bootstrap.runInternal(Bootstrap.java:163)
	at org.infinispan.server.tool.Main.run(Main.java:98)
	at org.infinispan.server.Bootstrap.main(Bootstrap.java:50)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.infinispan.server.loader.Loader.run(Loader.java:106)
	at org.infinispan.server.loader.Loader.main(Loader.java:51)
	Suppressed: java.net.UnknownHostException: 0.0.0.0 : Name or service not known
		at java.base/java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method)
		at java.base/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:929)
		at java.base/java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1519)
		at java.base/java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:848)
		at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1509)
		at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1368)
		at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1302)
		at java.base/java.net.InetAddress.getByName(InetAddress.java:1252)
		at org.infinispan.server.network.NetworkAddress.inetAddress(NetworkAddress.java:145)
		at org.infinispan.server.network.NetworkAddress.fromString(NetworkAddress.java:101)
		at org.infinispan.server.configuration.InterfaceConfigurationBuilder.createNetworkAddress(InterfaceConfigurationBuilder.java:43)
		at org.infinispan.server.configuration.InterfaceConfigurationBuilder.address(InterfaceConfigurationBuilder.java:23)
		at org.infinispan.server.configuration.ServerConfigurationParser.parseInterface(ServerConfigurationParser.java:218)
		... 22 more
```

File: README.md
Comment (tristantarrant) [reply README.md:126]: Fixed with the latest commit from https://github.com/infinispan/infinispan/pull/9676

File: modules/org/infinispan/runtime/added/bin/launch.sh
Comment (tristantarrant) [reply modules/org/infinispan/runtime/added/bin/launch.sh]: I've removed my changes

File: modules/org/infinispan/runtime/added/bin/launch.sh
Comment (tristantarrant) [reply modules/org/infinispan/runtime/added/bin/launch.sh:65]: Fixed

File: modules/org/infinispan/runtime/added/bin/launch.sh
Comment (ryanemerson) [inline modules/org/infinispan/runtime/added/bin/launch.sh:91]: We need to use the `-r` flag here, or provide a way for it to be explicitly configured on the image. IMO having a sensible default of `5` retries is sufficient for the Operator use-case.

Comment (ryanemerson) [reply modules/org/infinispan/runtime/added/bin/launch.sh:91]: We also need `--clean`

**Comment by tristantarrant** [comment]
Just pass the switches you want in the environment variable

`-e SERVER_LIBS="--clean -r 3 org.postgresql:postgresql:42.3.1"`

**Review by ryanemerson — APPROVED** [review]
LGTM. Just waiting on the server release before merging

**Pull request merged**
